vsdkx/addon/group/clustering.py

```diff
@@ -335,19 +415,47 @@ def post_process(self, addon_object: AddonObject) -> AddonObject:
             features, centroids, boxes = self.get_features(boxes,
                                                            trackable_objects)
-            self.cluster = self.DBSCAN_clustering(
-                centroids=features[:, temporal_data - 2:temporal_data],
-                distance_threshold_update=True)
+
+            if self.clustering_algorithm == 'dbscan':
+                self.cluster = self.dbscan_clustering(
+                    centroids=features[:, temporal_data - 2:temporal_data],
+                    distance_threshold_update=True)
+                # Cluster the centroids
+                y = self.cluster.fit(features)
+                # Separate them into groups > self.min_group_size
+                groups, people_count = self.get_groups(boxes,
+                                                       y.labels_,
+                                                       centroids)
+
+            elif self.clustering_algorithm == 'agglomerative':
+                self.cluster = self.agglomerative_clustering(
+                    centroids=features[:, temporal_data - 2:temporal_data],
+                    distance_threshold_update=True)
+                # Cluster the centroids
+                y = self.cluster.fit(features)
+                # Separate them into groups > self.min_group_size
+                groups, people_count = self.get_groups(boxes,
+                                                       y.labels_,
+                                                       centroids)
+
+            elif self.clustering_algorithm == 'chinesewhispers':
+                self.update_distance_threshold(
+                    centroids=features[:, temporal_data - 2:temporal_data])
+
+                print(f'Length of detected boxes {len(centroids)}'
+                      f' length of trackable objects {len(trackable_objects)}')
+                # Cluster the centroids
+                G = nx.from_numpy_matrix(features)
+                chinese_whispers(G, seed=1337)
+                y = aggregate_clusters(G)
+                # Separate them into groups > self.min_group_size
+                labels = self.extract_cluster_ids(y)
+                groups, people_count = self.get_groups(boxes,
+                                                       labels,
+                                                       centroids)
 
             print(f'Length of detected boxes {len(centroids)}'
                   f' length of trackable objects {len(trackable_objects)}')
-            # Cluster the centroids
-            y = self.cluster.fit(features)
-            # Separate them into groups > self.min_group_size
-            groups, people_count = self.get_groups(boxes,
-                                                   y.labels_,
-                                                   centroids)
-
         addon_object.inference.extra['tracked_groups'] = groups
         addon_object.inference.extra['objects_in_groups'] = people_count
 
```

Comment:
I think 461 lines for a single class is too much. If we merge this now, it's gonna need refactoring on next time we need to change this file, if we have such need.